src/components/SignalGeneratorComponent.cpp

```diff
@@ -0,0 +1,125 @@
+/**
+* @file SignalGeneratorComponent.cpp
+* @brief This file contains implementation for the definitions for the SignalGeneratorComponent functionalities.
+*
+* @author Murilo Mendonça Venâncio
+* 
+* @date 24/01/2019
+*/
+
+#include <hyro/SignalGeneratorComponent.hpp>
+
+namespace hyro
+{
+
+std::shared_ptr<hyro::HyroLogger> SignalGeneratorComponent::s_logger = hyro::HyroLoggerManager::CreateLogger("SignalGeneratorComponent");
+
+SignalGeneratorComponent::SignalGeneratorComponent (URI uri): hyro::Component(uri)
+{
+  /** Initialize object here */
```

Comment:
Your Constructor is doing nothing. I suggest you remove it from here and declare `using Component::Component` in class declaration.